perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_attribute_def_def_elixirBonaFideStatus.java

```diff
@@ -0,0 +1,115 @@
+
+	private static final String FRIENDLY_NAME = "elixirBonaFideStatus";
+	private static final String URL = "http://www.ga4gh.org/beacon/bonafide/ver1.0";
+
+	private static final String USER_BONA_FIDE_STATUS_REMS_ATTR_NAME = "elixirBonaFideStatusREMS";
+	private static final String USER_AFFILIATIONS_ATTR_NAME = "eduPersonScopedAffiliations";
+	private static final String USER_PUBLICATIONS_ATTR_NAME = "publications";
+
+	private static final String A_U_D_userBonaFideStatusRems = AttributesManager.NS_USER_ATTR_DEF + ":" + USER_BONA_FIDE_STATUS_REMS_ATTR_NAME;
+	private static final String A_U_D_userPublications = AttributesManager.NS_USER_ATTR_VIRT + ":" + USER_PUBLICATIONS_ATTR_NAME;
+	private static final String A_U_V_userAffiliations = AttributesManager.NS_USER_ATTR_VIRT + ":" + USER_AFFILIATIONS_ATTR_NAME;
+
+	private static final String ELIXIR_KEY = "ELIXIR";
+
+
+	@Override
+	public Attribute fillAttribute(PerunSessionImpl sess, User user, AttributeDefinition attributeDefinition) throws InternalErrorException, WrongAttributeAssignmentException {
+		Attribute attribute = new Attribute(attributeDefinition);
+
+		//try to get value from 'elixirBonaFideStatusREMS': if not empty, we have bona_fide
+		try {
+			Attribute statusAttr = sess.getPerunBl().getAttributesManagerBl().getAttribute(sess, user, A_U_D_userBonaFideStatusRems);
+			if (statusAttr != null && statusAttr.getValue() != null) {
+				attribute.setValue(URL);
+			}
+		} catch (WrongAttributeAssignmentException | AttributeNotExistsException e) {
+			log.error("Cannot read {} from user {}", USER_BONA_FIDE_STATUS_REMS_ATTR_NAME, user, e);
+		}
+
+		//try to get value from 'eduPersonScopedAffiliations': if has faculty@..., we have bona_fide
+		try {
+			Attribute affiliationsAttr = sess.getPerunBl().getAttributesManagerBl().getAttribute(sess, user, A_U_V_userAffiliations);
+			if (affiliationsAttr != null && affiliationsAttr.getValue() != null) {
+				ArrayList<String> affiliationsValue = affiliationsAttr.valueAsList();
+				for (String val : affiliationsValue) {
+					if (val.startsWith("faculty@")) {
+						attribute.setValue(URL);
+						return attribute;
+					}
+				}
+			}
+		} catch (WrongAttributeAssignmentException | AttributeNotExistsException e) {
+			log.error("Cannot read {} from user {}", USER_AFFILIATIONS_ATTR_NAME, user, e);
+		}
+
+		//try to get value from publications: if has thanks to ELIXIR, we have bona_fide
+		try {
+			Attribute publicationsAttr = sess.getPerunBl().getAttributesManagerBl().getAttribute(sess, user, A_U_D_userPublications);
+			if (publicationsAttr != null && publicationsAttr.getValue() != null) {
+				Map<String, String> publications = publicationsAttr.valueAsMap();
+				if (publications.containsKey(ELIXIR_KEY)) {
+					String value = publications.get(ELIXIR_KEY);
+					try {
+						int count = Integer.parseInt(value);
+						if (count > 0) {
+							attribute.setValue(URL);
```

Comment:
Will be more clear if there will be also return after setting value. Even if global return is right after this place.